Tests/SwiftKueryORMTests/TestSave.swift

```diff
@@ -42,19 +52,30 @@ class TestSave: XCTestCase {
         })
     }
 
+    /**
+      Testing that the correct SQL Query is created to save a Model
+      Testing that an id is correcly returned
+    */
     func testSaveWithId() {
         let connection: TestConnection = createConnection(.returnOneRow)
         Database.default = Database(single: connection)
         performTest(asyncTasks: { expectation in
             let person = Person(name: "Joe", age: 38)
             person.save { (id: Int?, p: Person?, error: RequestError?) in
                 XCTAssertNil(error, "Save Failed: \(String(describing: error))")
-                XCTAssertNotNil(connection.query, "Find Failed: Query is nil")
+                XCTAssertNotNil(connection.query, "Save Failed: Query is nil")
                 if let query = connection.query {
-                  let expectedQuery1 = "INSERT INTO People (name, age) VALUES ('Joe', 38)"
-                  let expectedQuery2 = "INSERT INTO People (age, name) VALUES (38, 'Joe')"
+                  let expectedPrefix = "INSERT INTO People"
+                  let expectedSQLStatement = "VALUES"
+                  let expectedColumns = ["name", "age"]
+                  let expectedValues = ["'Joe'", "38"]
                   let resultQuery = connection.descriptionOf(query: query)
-                  XCTAssert(resultQuery == expectedQuery1 || resultQuery == expectedQuery2)
+                  XCTAssertTrue(resultQuery.hasPrefix(expectedPrefix))
+                  XCTAssertTrue(resultQuery.contains(expectedSQLStatement))
+                  for (column, value) in zip(expectedColumns, expectedValues) {
```

Comment:
Here we are saying that the names and values can appear at any place in the query string and in any order.  This is a lot less stringent than the previous implementation.
We _could_ argue that, as the implementation is to give Kuery a list of tuples of column name and value, it's Kuery that is responsible for (for example) not mixing up the order and trying to assign values to the wrong columns, however we should at least state that as an assumption as a comment.

Reply:
I will revert it to the two permutations because it's the ORM which makes sure that the column -value pairs match.

Reply:
Another approach (though requiring some effort) would be to parse the generated SQL back into key and value arrays, zip them, and check each expected pair exists.

Reply:
Reges: .*\((.*)\)[^\(\)]*\((.*)\)

Reply:
Why are we not using the same (regex) approach as `testSave()` here?